src/funjson.c

```diff
@@ -64,6 +65,27 @@ json_escape_string(char *input)
   return buff;
 }
 
+/** Unescape a string for use as a JSON string. Returns a calloced buffer under 'json.string.partialbuf'. **/
+char * json_unescape_latin1string(char *latin1, int len)
+{
+  char *c;
+  char *partialbuf = mush_calloc(len*6,sizeof(char),"json.string.partialbuf");
+  char *bpp = partialbuf;
+  for (c = latin1; *c; c += 1) {
+    if(UNSAFE_UNESCAPE && (*c == ESC_CHAR || *c == TAG_END || *c == TAG_START )) {
```

Comment:
If you're going to protect markup, you should protect the entire markup string; from ESC_CHAR to 'm' and from TAG_START to TAG_END

Reply:
Can you explain this a bit further? There is no real protection here aside from the config option.

Reply:
It doesn't matter so much for ANSI (where the characters inside the tag will always be ASCII anyway), but for Pueblo or any other potential future markups there may be non-latin1 characters used. Depending on what kind of tag it is, non-printing characters may appear, and if you're keeping the markup you should really preserve everything in it to ensure it's correct.

Reply:
Gotcha. So just verbatim copy it all if UNSAFE_UNESCAPE.